pkg/expr/types.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expr
+
+import "istio.io/mixer/pkg/attribute"
+
+type (
+	// Evaluator maps Given an expression Language
+	Evaluator interface {
+		// Eval evaluates given expression using the attribute bag
+		Eval(mapExpression string, ctx attribute.Bag) (interface{}, error)
+
+		// Eval evaluates given expression using the attribute bag to a string
+		EvalString(mapExpression string, ctx attribute.Bag) (string, error)
```

Comment:
Here and elsewhere, I think we should be using some other name instead of "ctx". We'll eventually be plumbing the context.Context stuff through and that should be called "ctx" by convention. So perhaps, the bags should be called "bag" or "attrs".